.github/workflows/dependency_review.yaml

```diff
@@ -0,0 +1,101 @@
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+      - name: Dependency Security Review
+        uses: actions/dependency-review-action@v2
+        with:
+          fail-on-severity: low
+          allow-licenses: MIT
+  dependency-updates:
+    name: Versions
+    runs-on: ubuntu-latest
+    steps:
+      - name: Environment
+        run: env
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+      - name: Dependency Versions Review
+        id: dependency_versions_review
+        run: |
+          echo "::notice ::BX_PR_NUMBER=${BX_PR_NUMBER}"
+
+          # error off since we use 'git diff' to determine changes
+          set +o errexit
+          echo "::group::Downloading dependencies"
+          go get -u all; go mod tidy
+          echo "::endgroup::"
+          set -o errexit
+
+          TMP_GIT_STATUS_GO_MOD=$(git diff go.mod)
+          echo "::debug ::TMP_GIT_STATUS_GO_MOD=${TMP_GIT_STATUS_GO_MOD}"
+
+          if [ "${TMP_GIT_STATUS_GO_MOD}" == "" ]; then
+            echo "::warning ::Dependencies are NOT outdated"
+            echo "::set-output name=dependencies_outdated::true"
+
+            else
+              echo "::warning ::Dependencies are outdated"
+              echo "::set-output name=dependencies_outdated::true"
+          fi
+        shell: bash
+      - name: Request PR Changes
+        if: ${{ steps.dependency_versions_review.outputs.dependencies_outdated == 'true' }}
+        uses: actions/github-script@v6
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const res = await github.rest.pulls.listReviewComments({
+              owner: context.repo.owner,
+              pull_number: context.issue.number,
+              repo: context.repo.repo
+            })
+            console.log("Received response", res)
+
+            let foundComment = false
+            if (res && res.data) {
+              console.log("Found existing comments, will check to see if 'DEPENDENCIES OUTDATED' has already been made", res.data)
+              for (const d of res.data) {
+                if (d.body && d.body === "🔴 DEPENDENCIES OUTDATED 🔴") {
+                  foundComment = true
+                  break
+                }
+              }
+            }
+
+            if (foundComment) {
+              console.log("Found existing 'DEPENDENCIES OUTDATED' comment, will not add review", res.data)
+            }
+            else {
+              console.log("Existing 'DEPENDENCIES OUTDATED' comment not found, will add review")
+
+              github.rest.issues.addLabels({
+                issue_number: context.issue.number,
+                labels: ["🔴 DEPENDENCIES OUTDATED 🔴"],
+                owner: context.repo.owner,
+                repo: context.repo.repo
+              })
+
+              github.rest.pulls.createReview({
+                body: "🔴 DEPENDENCIES OUTDATED 🔴",
+                event: "REQUEST_CHANGES",
```

Comment:
I guess this is sensible, then the PR owner has to manually dismiss that review in order to merge the commit.

But I can see a scenario where the first commit on the branch has outdated deps, but the final commit has deps updated, yet this change request will still be here

Reply:
Sure there are lots of edge cases but this is manual and so the owner can make a judgment call

Reply:
yeah, it is a timeline too so it will be clear that it is not necessarily the truth if subsequent commits exist